src/compiler/checker.ts

```diff
@@ -13011,6 +13282,69 @@ namespace ts {
             return resolveCall(node, callSignatures, candidatesOutArray, headMessage);
         }
 
+        /**
+         * 
+         * @param openingLikeElement
+         * @param elementType
+         * @param candidatesOutArray
+         */
+        function getResolvedJSXStatelessFunctionSignature(openingLikeElement: JsxOpeningLikeElement, elementType: Type, candidatesOutArray: Signature[]): Signature {
+            Debug.assert(!(elementType.flags & TypeFlags.Union));
+            const links = getNodeLinks(openingLikeElement);
+            // If getResolvedSignature has already been called, we will have cached the resolvedSignature.

```

Comment:
This code looks a *lot* like `getResolvedSignature`. Are you sure you can't re-use `getResolvedSignature` and move the core difference either down in to `resolveStatelessJsxOpeningLikeElement` or up into `tryGetJsxStatelessFuncitonAttributesType`?